src/tests/spells/red.ts

```diff
@@ -1,10 +1,93 @@
 import { expect } from "chai";
 
+import { BASE_CARD } from "../../data/core";
+import { GameEngine } from "../../framework/engine";
+import { createInstance } from "../../framework/mutators";
+import { PlayerID, PlayerState } from "../../framework/types";
+import { requireActivePlayer } from "../../game/helpers";
+
+import {
+  P2,
+  debugAction,
+  debugPlayCard,
+  makeDefaultGame,
+} from "../testhelper";
+
 describe("spells", () => {
   describe("red", () => {
+    let $: GameEngine;
+    let P: PlayerState;
+    let oppP: PlayerState;
+
+    beforeEach(() => {
+      $ = makeDefaultGame();
+      P = requireActivePlayer($);
+      oppP = $.state.players[P2];
+    });
+
     describe("Pillage", () => {
+      beforeEach(() => {
+        const EX_RED_HERO = $.data.lookupCard("Captain Zane");
+        createInstance($, P, EX_RED_HERO);
+      });
+
+      const playAndResolvePillage = (
+        targetPlayer: PlayerID,
+      ): void => {
+        debugPlayCard($, "Pillage");
+        expect($.state.unresolvedEffects.length).to.equal(1);
+        const eid = $.state.unresolvedEffects[0].id;
+
+        const base = $.findInstance({
+          card: BASE_CARD.id,
+          player: targetPlayer,
+        });
+        if (base == null) {
+          throw new Error("failed to find base for pillage");
+        }
+
+        debugAction($, {
+          type: "RESOLVE_EFFECT",
+          effect: eid,
+          params: {
+            target: base.id,
+          },
+        });
+      };
+
       it("can't steal more gold than the target has", () => {
-        expect(true).to.equal(false);
+        P.gold = 0;
+        oppP.gold = 0;
+        playAndResolvePillage(oppP.id);
+        expect(P.gold).to.equal(0);
+        expect(oppP.gold).to.equal(0);
+
+        P.gold = 0;
+        oppP.gold = 1;
+        playAndResolvePillage(oppP.id);
+        expect(P.gold).to.equal(1);
```

Comment:
Technically you don't have a case for stealing 1 gold from an opponent with 2, though you do cover taking 2 from an opponent with 3...

Reply:
Yeah, I didn't feel like that mattered. I just wanted to verify in-line that with the pirate (which is the most tested state) we were attempting to steal 2 always.